r/src/differential_expression.r

```diff
@@ -166,31 +43,28 @@ runDE <- function(req){
     # add seurat object the new groups to compare    
     data@meta.data$custom <- NA
     data@meta.data[names(factor_DE), "custom"] <- factor_DE
-    
-    # compute differential expression
-    result <- compute_Zscore(data@assays$RNA@data, groups = factor_DE)$base
 
-    # replace name with Gene names
+    # Compute differential expression
+    result <- FindMarkers(data, group.by = "custom", ident.1 = "base", ident.2 = "background")
+    # Replace name with Gene names
     result$gene_names <- data@misc$gene_annotations[
         match(rownames(result), data@misc$gene_annotations$input), "name"
     ]
-    
-    # Alternative with seurat DE pipeline
-    # result <- FindMarkers(data, group.by = "custom", ident.1 = "base", ident.2 = "background")
-    # # replace name with Gene names
-    # result$gene_names <- data@misc$gene_annotations[
-    #    match(rownames(result), data@misc$gene_annotations$input), "name"
-    #]
-    # result$Gene <- rownames(result)
-    # result$p_val <- NULL
-    # colnames(result) <- c("log2fc", "pct.1", "pct.2", "p_val_adj", "gene_names", "Gene")
+    result$Gene <- rownames(result)
+    result$p_val <- NULL
 
-    return(result)
-}
+    # As a first view, order by p_val_adj, to have the most significant at first.
+    result <- result[order(result$p_val_adj, decreasing = F), ]
 
+    # Change "." in pct.1 by _
+    colnames(result) <- gsub("[.]", "_", colnames(result))
```

Comment:
Nice! Didn't know that square brackets makes metachars explicit. Alternative would be `gsub(..., fixed = TRUE)` but I like this :)